src/imgui/ImGuiReverseBar.cc

```diff
@@ -62,10 +62,32 @@ void ImGuiReverseBar::showMenu(MSXMotherBoard* motherBoard)
 		}
 		ImGui::Separator();
 
+		auto fileTimeToString = [](std::filesystem::file_time_type fileTime) {
+			// As we still want to support gcc-11 (and not require 13 yet), we have to use this method
+			// as workaround instead of using more modern std::chrono and std::format stuff.
+
+			// Convert file_time_type to system_clock::time_point (the standard clock since epoch)
+			auto sctp = std::chrono::time_point_cast<std::chrono::system_clock::duration>(
+					fileTime - std::filesystem::file_time_type::clock::now() +
+					std::chrono::system_clock::now());
+
+			// Convert system_clock::time_point to time_t (time since Unix epoch in seconds)
+			std::time_t cftime = std::chrono::system_clock::to_time_t(sctp);
+
+			// Convert time_t to local time (broken-down time in the local time zone)
+			std::tm* local_time = std::localtime(&cftime);
+
+			// Get the local time in human-readable format
+			std::stringstream ss;
+			ss << std::put_time(local_time, "%F %T");
+			return ss.str();
+		};
```

Comment:
Good idea to factor this out.

This is also a good place to implement the idea of "show exactly one of date or time, depending on whether date==today". It'll help keep the sub menu smaller.  (For sure `std::chrono` offers a way to extra the date part from a time-point, including from `system_clock::now()`. But you'll have to google a bit.)

Nitpick: I don't know if this is a universal convention. But I prefer the name `xxxToString` when there's a (semi-)unique string-representation for `xxx`. And the name `formatXxx` when this function chooses one of many possible formattings.

Reply:
I'm not sure it's a good idea to do this actually. If you have saved many states/replays on one day, but that was yesterday, you can't even see anymore the times at which you saved them...

Reply:
I didn't repeat this part, but the idea was that if you hover over the date, it still shows the full date + time.

Reply:
Alright, so I tried to make something like this:
```
		auto formatFileTime = [](std::filesystem::file_time_type fileTime) {
			// As we still want to support gcc-11 (and not require 13 yet), we have to use this method
			// as workaround instead of using more modern std::chrono and std::format stuff.

			// Convert file_time_type to system_clock::time_point (the standard clock since epoch)
			auto sctp = std::chrono::time_point_cast<std::chrono::system_clock::duration>(
					fileTime - std::filesystem::file_time_type::clock::now() +
					std::chrono::system_clock::now());

			// Convert system_clock::time_point to time_t (time since Unix epoch in seconds)
			std::time_t cftime = std::chrono::system_clock::to_time_t(sctp);

			// Convert time_t to local time (broken-down time in the local time zone)
			std::tm* local_time = std::localtime(&cftime);

			std::time_t t_now = std::time(0);   // get time now
			std::tm* now = std::localtime(&t_now);

			const std::string format = ((now->tm_mday == local_time->tm_mday) &&
						    (now->tm_mon  == local_time->tm_mon ) &&
						    (now->tm_year == local_time->tm_year)) ? "%T" : "%F";

			// Get the local time in human-readable format
			std::stringstream ss1, ss2;
			ss1 << std::put_time(local_time, format.c_str());
			ss2 << std::put_time(local_time, "%F %T");
			return std::pair{ss1.str(), ss2.str()};
		};
```
But for some reason totally unclear to me, I only get the current date/time as results.... Even though I'm putting `local_time`'s value in the `stringstream`s.  How can querying "now" affect the results of `local_time`?

Reply:
`localtime()` is a C-function (even if it's made available in the c++ `std` namespace). It returns a _pointer_ to some `tm` struct. Calling `localtime()` a second time may return the same pointer (and thus overwrite the data in the same struct).

Possible solutions are:
* Replace `localtime()` with `localtime_r()`, in the latter you pass a pointer to your own `struct tm` that will be filled in. Though it's unclear how "portable" this function is (it works on Linux, but I'm not sure about Windows).
* Make a copy of the returned struct: `std::tm tm1 = *std::localtime(...);`

Reply:
In this prototype returning both formats is fine. In the final version I think it might be better to split this into two functions.
Reasoning: most of the time you only need one of these two: on most frames the tooltip isn't visible, And then it's wasteful to eagerly calculate both.
And yes, in case you do need both then you redo part of the calculation, but that's a minority of the cases.

Reply:
Done, thanks for all the suggestions and hints.